### ISSUE TITLE
GUI build script rewrite

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#4410]
This task is automatically imported from the old Task Issue Board and it was originally created by Wojciech Daniło.
Original issue is [here](https://www.pivotaltracker.com/story/show/180302898).
<hr />

## Description
Rewrite the JS scripts developed for GUI. Integrate already written Rust scripts.

## Resources
This involves primarily creating a functional equivalent to the `run` JS script that was developed for GUI.

## Comments:
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-02-15):

**Today:** Mostly finishing maintenance and cleanups in preparation for the new build script. Deployed the current version of the build script and addressed the resulting bumps. Preparing presentation for demo day. Weekly epic update call. James participated, we discussed possible ways to speed up the CI. It should be finished by 2022-03-02.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Get to the unified IDE + Engine package building. For real. (Enso Bot - Feb 16, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for yesterday (2022-02-21):

**Summary:** There is **5 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 5 days delay** for the delivery of this sprint.

Includes weekend.

**Delay Cause:** I was focusing on optimizing PR workflow. (Enso Bot - Feb 22, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-02-21):

**Today:** Finally properly got back to the IDE build script. Implemented a new approach for handling with paths, rewritten signifact parts of the top script layers.
Mitigited minor malady with mergify and CLA assistant. It should be finished by 2022-03-07.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue work on build script rewrite. Prepare weekly epic presentation. (Enso Bot - Feb 22, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-02-22):

**Today:** Prepared and presented the weekly epic team update.
Working on the outer layers: building the Rust, preparing and invoking nested packages builds. It should be finished by 2022-03-07.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue work on build script rewrite. (Enso Bot - Feb 23, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-02-23):

**Today:** For the first time generated a unified package with somewhat newer engine and latest GUI. Bringing myuself up-to-date with the current build scheme of IDE and JS tech stack. It should be finished by 2022-03-07.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Refactor, prepare for semi-manual building of unified local/local package. (Enso Bot - Feb 24, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-02-24):

**Today:** Refactoring Engine build script to prepare it for the integration with the unified script. It should be finished by 2022-03-07.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Attempt to make semi-automatic local build of the unified local/local package. (Enso Bot - Feb 25, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-02-25):

**Today:** Continues refactorings. To be honest, not much progress was made, I've found myself quite distracted by the war news. It should be finished by 2022-03-07.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Attempt to make semi-automatic local build of the unified local/local package. (Enso Bot - Feb 28, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-02-28):

**Today:** Some progress on generating local/local package. Abstracting the project manager provider, so IDE can be easily built either with local PM or with a one from a release.
Looking into parallelizing the GUI/Engine build. It should be finished by 2022-03-07.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Complete the experimental local/local package. (Enso Bot - Mar 1, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for yesterday (2022-03-01):

**Summary:** There is **4 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 4 days delay** for the delivery of this sprint.

**Delay Cause:** My velocity is low, JS gives me surprises, packages are slow to build. (Enso Bot - Mar 2, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-01):

**Today:** Build local/local package for the first time, though with some some manual intervention and dirty changes. It should be finished by 2022-03-10.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Cleanups so the package builds with less trouble. (Enso Bot - Mar 2, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-02):

**Today:** Very little actual progress. I've tried a few approaches to stick together the JS parts and Rust but they somehow did not feel right. Well, let's call it "research". It should be finished by 2022-03-10.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Cleanups so the package builds with less trouble. (Enso Bot - Mar 3, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-03):

**Today:** Still very little progress, trying to understand how the JS, Electron, Webpack, Lerna and NPM work together. Had a calls and discussed possible approaches with Wojciech and Adam. While I haven't significantly moved forward implementation-wise, now I feel much better about my understanding and design. It should be finished by 2022-03-10.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Cleanups so the package builds with less trouble. (Enso Bot - Mar 4, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-03-04):

**Today:** Ramping up the actual work. Improvements in the icon generation script, so it is more consistent and self-reliant. Extracted Electron Builder config into a separate file. With this, we can remove the `package.json` generation step. Added support for using project manager from the specified local path in the new build script. It should be finished by 2022-03-10.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue the above. Refactor the JS part of the project while developing the corresponding support on the new build script side. (Enso Bot - Mar 7, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for yesterday (2022-03-07):

**Summary:** There is **7 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 7 days delay** for the delivery of this sprint.

**Delay Cause:** I have vastly underestimated the JS trickery. As I delved into the codebase, debugged various issues and discussed design with Wojciech, the need for deeper refactorings in the JS project part became apparent. (Enso Bot - Mar 8, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-07):

**Today:** Tested packages for Dmitry. Writing down issues and comparing regressions against the alpha 18 release.
Debugging issues within the IDE that were exposed by the new versioning scheme. Fixing the minimum supported version check that was broken. It should be finished by 2022-03-17.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue the above. Start moving the patched up things into repos, so the CI can try building the packages. (Enso Bot - Mar 8, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-08):

**Today:** Preparing weekly update and presenting it.
Getting rid of lerna. Cleanups and JS-related experiments. It should be finished by 2022-03-17.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue the above. Start moving the patched up things into repos, so the CI can try building the packages. (Enso Bot - Mar 9, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-09):

**Today:** Just endless stream of cleanups, moving things around and fixes, while repeatedly building the local package.
Reviewing and discussing the enso code formatter. It should be finished by 2022-03-17.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue the above. If things go good, attempt CI calls to build packages. (Enso Bot - Mar 10, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-10):

**Today:** Started work to build packages in actual CI environment. Got self-hosted Linux and Windows builds to pass. It should be finished by 2022-03-17.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Work on getting macOS package to build. Upload the generated packages. (Enso Bot - Mar 11, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-03-11):

**Today:** Worked on fixing the macOS packages build. Encountered issues with signing. When I was trying to debug it, I've encountered issues with path containing space. It should be finished by 2022-03-17.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue to work on delivering macOS package for initial tests. (Enso Bot - Mar 15, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for today (2022-03-14):

**Summary:** There is **4 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 4 days delay** for the delivery of this sprint.

(incl weekend)

**Delay Cause:** macOS costed me some time that I've not expected. Also, it seems that some debugging will be necessary. (Enso Bot - Mar 15, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-03-14):

**Today:** Demo day and weekly planning meeting. 
As per discussion, the macOS package signing was temporarily disabled to get the package building.
I've succeeded in delivering the first macOS package, however early testing shows some issues (likely some engine version mismatch in the generated project). It should be finished by 2022-03-21.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Debug the package issue, work on improving its build reliability/ (Enso Bot - Mar 15, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-15):

**Today:** Much of the day was spent on Windows code signing certificate having expired on Saturday — this required figuring what has happened, applying workarounds and starting process for getting a new certificate.
Thanks to Ola and Sylwia for help with the process. Thanks to Radek and Kaz for getting my attention to the problem.

Otherwise, still trudging on better build script behavior. Cleanups to enable building IDE with arbitrary sources of wasm artifacts. It should be finished by 2022-03-21.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Back to the packaging, continue the above. (Enso Bot - Mar 16, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Wednesday (2022-03-16):

**Progress:** Implemented different wasm sources. Split web-content target from the rest. Started refactoring how program calls are handled. It should be finished by 2022-03-21.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Packaging. (Enso Bot - Mar 18, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-17):

**Progress:** Refactored the program commands handling, to enable providing option wrappers in a more composable way than before.  Cleanup the interface between build script and JS content package. Started work on a new entry point. It should be finished by 2022-03-21.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). New entry point. Fix the engine version used by the IDE when creating a new project. (Enso Bot - Mar 18, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for today (2022-03-21):

**Summary:** There is **7 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 0 days delay** for the delivery of this weekly plan.

(incl weekend)

**Delay Cause:** One more week will be needed. macOS is still causing troubles, unexpected CI incidents took some time and packaging always is more time consuming than I'm able to predict. Sigh. (Enso Bot - Mar 22, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-03-18):

**Progress:** Prepared a new entry point, set up new CI jobs to use it. So far only WASM has been completed. Refactorings as to per how new build tool is provided to GH Actions to reduce boilerplate.
Had a discussion with Adam, incl. the topic of the default engine for the newly created project. It should be finished by 2022-03-28.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue developing new CLI driver and CI. (Enso Bot - Mar 22, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-03-18):

**Progress:** Completed IDE Contents CI job. Trying to handle macOS-specific (technically, GH-hosted runner-specific) CI boilerplate in a smarter way. 
Weekly presentation / planning call. It should be finished by 2022-03-28.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue developing new CLI driver and CI. (Enso Bot - Mar 22, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-03-18):

**Progress:** Completed IDE Contents CI job. Trying to handle macOS-specific (technically, GH-hosted runner-specific) CI boilerplate in a smarter way. 
Weekly presentation / planning call.
Developing new CLI driver, reusing groups of flags between subcommands. Allowing building specific targets while specifying how the dependencies are to be obtained. It should be finished by 2022-03-28.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue developing new CLI driver and CI. Get back to fixing the Engine default version for the IDE-created projects. (Enso Bot - Mar 22, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-03-21):

**Progress:** Completed IDE Contents CI job. Trying to handle macOS-specific (technically, GH-hosted runner-specific) CI boilerplate in a smarter way. 
Weekly presentation / planning call.
Developing new CLI driver, reusing groups of flags between subcommands. Allowing building specific targets while specifying how the dependencies are to be obtained. It should be finished by 2022-03-28.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue developing new CLI driver and CI. Get back to fixing the Engine default version for the IDE-created projects. (Enso Bot - Mar 22, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-22):

**Progress:** Support for building project manager bundle through the new CLI driver. Some refactorings in the engine build script to allow for more selective builds. Started and achieved significant progress with building the full, unified IDE package through the new driver. It should be finished by 2022-03-28.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue developing new CLI driver and CI. (Enso Bot - Mar 23, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-23):

**Progress:** Finally got the unified package to correctly build from the new cli driver. It should be finished by 2022-03-28.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Work on multi-stage builds. Wire things with CI. (Enso Bot - Mar 24, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-24):

**Progress:** Hardly got anything done but I am kind of satisfied with progress. I've started a number of refactorings - how build targets are handled, how to deduplicate some code in the CLI arguments description, etc. It should be finished by 2022-03-28.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Finish the above. (Enso Bot - Mar 25, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-03-25):

**Progress:** Got to the point where IDE package builds again on the new CLI driver. Support for getting different components from different sources. It should be finished by 2022-03-28.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). … (Enso Bot - Mar 29, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for today (2022-03-28):

**Summary:** There is **14 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Reevaluation based on current velocity and scope. Basically while the progress is steady recently, it is nowhere near the velocity needed to wrap everything up in the next week. (Enso Bot - Mar 29, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-03-28):

**Progress:** Preparing the weekly update. 

Started setting the new IDE build up on the CI. Several issues discovered. In particular, building wasm and Enso in parallel turned to be disastrous, as the latter was relying on repository getting cleaned. There's an issue where paths in the uploaded artifacts are seemingly garbled. It should be finished by 2022-04-11.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Get the CI packages to be built. (Enso Bot - Mar 29, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-29):

**Progress:** Work on addressing issues with packaging and uploads. Not really done, but addressed most of the issues. It should be finished by 2022-04-11.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Get the CI packages to be built. (Enso Bot - Mar 30, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-30):

**Progress:** Addressed issues preventing the package build, gave a new package to Sylwia. 
There's seemingly issue in there where backend hangs with full cpu usage, it's unclear whether it is issue with packaging or engine itself. It should be finished by 2022-04-11.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Work on watch command. Investigate the high CPU usage in the engine. (Enso Bot - Mar 31, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-03-31):

**Progress:** Implemented `ide build` variant that automatically downloads project manager from a different run (specified through CLI) artifact and builds IDE against the downloaded package. Started work on watch command. It should be finished by 2022-04-11.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Work on watch command. (Enso Bot - Apr 1, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-04-01):

**Progress:** Implemented spawned process output intercepting and feeding into our own logging. This is important for watch where we spawn a few programs at the same time and their output is interleaved.
Tried to implement Rust-side watch on top of `watchexec` library directly but it turned out much more time consuming than I expected. So after some failed attempts, I decided to revert to the traditional cargo-watch-based approach. It should be finished by 2022-04-11.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Work on watch command. (Enso Bot - Apr 5, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-04-04):

**Progress:** Restored the watch command built on top of cargo watch. 
Refactoring the argument-handling module, splitting it into multiple project-specific submodules. 
Implemented `check` and `test` commands for their respective projects. It should be finished by 2022-04-11.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Finish watch command by adding project manager spawning in the background. (Enso Bot - Apr 5, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-05):

**Progress:** Had a weekly update meeting with Wojciech and Sylwia, this time in the office. (don't ask why Monday is on Tuesday)
Dealing with a number of minor stuff: testing "nvm for windows" (bad), quick research for code signing certificate providers and macOS hosting. Built new packages with updated engine.
Implemented spawning of project manager. However, it is buggy. It should be finished by 2022-04-11.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Test and polish watch command, (Enso Bot - Apr 6, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-06):

**Progress:** Done some research on the `tracing` library and its WASM backend. After discussion with Wojciech we decided to postpone any implementation work on this, until we have a better understanding how we want to approach future logging framework.
Completed feature-wise  `watch` command (incl. watching a specific crate, missing cargo custom flags that are more general feature). Implemented `clean` command. It should be finished by 2022-04-11.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Test and polish `watch` command. Support for passing cargo arguments for watch and build. (Enso Bot - Apr 7, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-07):

**Progress:** Improved output handling for NPM when invoked by watch command. Refactored handling of environment variables for sharing data with JS scripts. Bumped WebPack with follow-up fixes. Refactored some duplication from build and dist commands. It should be finished by 2022-04-11.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Found issue with dist command, need to fix it. There seems to be some race condition in the watch command -- look into it. (Enso Bot - Apr 8, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-04-08):

**Progress:** Implemented passing arbitrary cargo options when building/watching. A few more fixes following the webpack bump. Some improvements to watch output. It should be finished by 2022-04-11.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Testing watch, filling the gaps in the supported commands. (Enso Bot - Apr 11, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for yesterday (2022-04-11):

**Summary:** There is **14 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Still low velocity development. High iteration times (builds), issues hard to debug (race conditions, issues on the brink of numerous external tools). Doing odd jobs in the meantime. Many unexpected issues turning up. (Enso Bot - Apr 12, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-11):

**Progress:** One of those days when everything is broken. Updated CI runners, only to encounter two regressions: `go get`  syntax stopped working and latest `npm` breaks things for Windows build. Had to re-deploy windows runners with older npm pinned.
As for build script, hardly any progress -- I've encountered an issue with building packages on CI that does not reproduce outside CI. Spent much time debugging it. It should be finished by 2022-04-25.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue debugging. Leave no stone unturned. (Enso Bot - Apr 12, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-12):

**Progress:** Debugging the issue from the day before. Eventually narrowed down the issue to the specific electron-notarize version bump. However, how and why it acts to delete necessary files during packaging remains mystery. No changes on the side of the notarizer explains this. For now I'm stopping further debugging, as we can just avoid the issue. It should be finished by 2022-04-25.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). (below) (Enso Bot - Apr 13, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-04-13):

**Progress:** Got back to the building script. Packages are building again. Looked into addressing the symlink preservation in GH artifacts. Unfortunately, this is impossible without introducing additional layer of archive (e.g. tar). This is really annoying, as GH packs the artifacts into ZIP once again, even if it was already an archive. 
Done some learning on the `tracing` library and started introducing it to the build script. It looks promising. Trying to improve script to keep track of ongoing processes — so it is visible when (and how) it is busy. Tried to achieve it with `indicatif` library, this seems somewhat tricky. It should be finished by 2022-04-25.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). More tracing and cleanups in how output is being handled. Limit possible venues of spawning programs without intercepting their output. Add workaround for race'y rustup behavior. (Enso Bot - Apr 13, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-14):

**Progress:** Improved logging. Fixed how the uploading directories as artifacts behaves. Fixed regression with generating checksums. Restored workarounds to reduce `watch` rebuilds. It should be finished by 2022-04-25.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Support for getting PM from release. (Enso Bot - Apr 15, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-18):

**Progress:** (this should be actually a Friday but it is was holiday on Monday, so posting it today)
Working on downloading project bundles from GH releases. Refactoring some related code. It should be finished by 2022-04-25.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Complete the above. (Enso Bot - Apr 19, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-19):

**Progress:** Implemented downloading downloading target bundles from the GitHub releases (primarily for the project manager). Had some trouble due to a bug in octocrab library. Adding related utilities to the archive-handling code. Instrumenting related code with logs. It should be finished by 2022-04-25.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Integrate getting PM from release with the CLI driver. Support for convenience functionalities (like downloading latest release). (Enso Bot - Apr 20, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-20):

**Progress:** Support for downloading specific PM release from GH assets by passing a relevant tag name or latest stable release or latest nightly release. It should be finished by 2022-04-25.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Adding more of the missing commands, caching PM downloads. (Enso Bot - Apr 21, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-21):

**Progress:** Implemented a basic filesystem cache and support for caching file downloads. Fixing some regressions with the package building. It should be finished by 2022-04-25.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Finish the caching for project manager. (Enso Bot - Apr 22, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-04-22):

**Progress:** Implemented caching for extracted archives. This covers the "get from the release" case for project manager acquisition — any given release asset should be downloaded once and be cached by default at a user level. It should be finished by 2022-04-25.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Update the Enso branch, get to work on integration tests. (Enso Bot - Apr 25, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for yesterday (2022-04-25):

**Summary:** There is **14 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Huge scope. (Enso Bot - Apr 26, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-25):

**Progress:** Updated branch on the Enso repository, resolved conflicts. Partially implemented WASM size check. Implemented check for wasm-pack version. Implemented selection of the wasm-pack build type (dev/release/profile). Implemented caching for extracted archives.The project manager distribution will be rather symlinked then copy when preparing a build. It should be finished by 2022-05-09.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Integration test command. Other various missing bits and pieces from the CLI interface. (Enso Bot - Apr 26, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-26):

**Progress:** Implemented integration-test command. Improvements for managing spawned project manager. Improved release handling. It should be finished by 2022-05-09.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue minor improvements for CLI, start experiments iwth enso-staging repository. (Enso Bot - Apr 27, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-04-27):

**Progress:** Reorganized the code to improve its usability as a library. Started preparing enso-staging with the new scripts. It should be finished by 2022-05-09.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Work on deploying to enso-staging. (Enso Bot - Apr 28, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-04-28):

**Progress:** Initial integration with the enso-staging repository. Testing on other platform. Looking into build script build time. Devising solution so the build script can be easily used without busting the IDE build caches. Wrapper scripts. It should be finished by 2022-05-09.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Rework the GUI CI on staging repository to rely on the new script. (Enso Bot - Apr 29, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-05-03):

**Progress:** This  report should be for Friday (29.04)
Adding support for lint and format commands on the integrated repo. It should be finished by 2022-05-09.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue CI rework on enso-staging: running tests. (Enso Bot - May 4, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-05-04):

**Progress:** Running IDE tests through CI. Various minor fixes. Did first test with a help of Adam, collected a number of issues. It should be finished by 2022-05-09.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Fix issue with broken integration test, add more missing jobs, address issues freported by Adam. (Enso Bot - May 5, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-05-05):

**Progress:** Fixing a number of bugs: prettier installation, integration test timeouts, passing additional cargo arguments for integration tests. Added CI jobs for separate WASM compilation + packaging per platform. It should be finished by 2022-05-09.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Debug more issues with integration tests. Look into addressing issue with PM recompilation. Refactor workflows. Work towards filling gaps in scripts. (Enso Bot - May 6, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-05-06):

**Progress:** Fixed integration tests. Implemented generating workflow description from Rust. Refactored and completed most of the missing workflows. Enabled build for all platforms. It should be finished by 2022-05-09.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Debug and work on the PM incremental build issue. Weekly update. Review the parser PR. (Enso Bot - May 9, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for yesterday (2022-05-09):

**Summary:** There is **14 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Huge scope. In general, progress is being tracked as part of weekly epic calls than this task. (Enso Bot - May 10, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-05-09):

**Progress:** Preparing and presenting weekly epic team update.
Fixed the issue with backend builds on self-hosted runners. Improvements and refactorings to the generated workflow definitions. It should be finished by 2022-05-23.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Fix the watch command issues. Gather more feedback about the script. (Enso Bot - May 10, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-05-10):

**Progress:** A number of calls (1-on-1 with Sylwia, discussing parser with Wojciech, coffee with Hubert and discussing build script with Adam).
I've engaged Michael into testing the new build script.
Various minor fixes to the script.
Fixing the issues with `watch` command where some CLI arguments were not correctly passed to the WASM build.
Improved ergonomics of CLI for reusing local build artifacts (it is not necessary to provide artifact path if it is same as default). It should be finished by 2022-05-23.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue collecting and addressing feedback about the new build script, with goal of  deployyment next week. (Enso Bot - May 10, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-05-11):

**Progress:** Visited the office, where together with Jaroslav we tried to debug the runtime's incremental compilation issue.
Fixes/improvements in build script, mostly centered around integration test. It should be finished by 2022-05-23.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). (below) (Enso Bot - May 12, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-05-12):

**Progress:** Fixes, improvements, cleanups, largely driven by feedback received. Added flag allowing running integration test with a browser. Improved CLI's help. It should be finished by 2022-05-23.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Look into support for cross-compilation (Michael's Linux->Windows case).
Update to follow up main repository changes. 
Continue to address feedback. (Enso Bot - May 12, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-05-13):

**Progress:** Working on supporting the cross-compilation feature that is used by Michael. This actually involved a somewhat bigger refactoring than initially expected. It should be finished by 2022-05-23.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). (below) (Enso Bot - May 16, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-05-16):

**Progress:** Weekly update, interactive review of parser with Wojciech.
Addressing feedback to the script: improving CLI help, imporving diagnostics for some error scenarios, fixing a race condition that can happen when `dist` directory does not exist when the build is started. It should be finished by 2022-05-23.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue addressing feedback and supporting people who are testing the script. (Enso Bot - May 16, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-05-17):

**Progress:** Addressing feedback to the script: Michael found an interesting problem where building a package with pre-built backend caused IDE to use wrong engine version. Solved this by adding two levels of default preferred engine version. One is within GUI itself (it will default to the one with the same version), and it can be override by IDE client — which will happen when IDE is bundled with backend. Additionally, exposed this option through client CLI interface, so it is possible to have IDE/Project Manager/Engine each in different version.
Restoring missing bits of the new GUI CI workflow.
Had a calls with Michael and Sylwia, regarding the coordination on deploying the new script with the ongoing profiler work and related CLI changes. It should be finished by 2022-05-23.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue addressing feedback and supporting people who are testing the script. Onboard more testers. (Enso Bot - May 19, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-05-18):

**Progress:** Fixing various minor issues / improvements. Addressing feedback from Adam, Mateusz and Kaz. Adding mechanism to prevent unnecessary calls to `gu install` as it might require root privileges. It should be finished by 2022-05-23.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue onboarding more testers and fixing reported problems. (Enso Bot - May 19, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for yesterday (2022-05-23):

**Summary:** There is **14 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Huge scope. In general, progress is being tracked as part of weekly epic calls than this task. (Enso Bot - May 25, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-05-23):

**Progress:** Preparing and presenting weekly epic team update. Working on fixes for issues observed following the build script deploy.
Fixing issue with mismatched paths on M1-based macOS systems. Fixing issue with mismatched client executable on Linux/mac. 
Debugging build issues on CI reported by Adam. It should be finished by 2022-06-06.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue with fixes, particularly the lint errors. (Enso Bot - May 25, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-05-24):

**Progress:** Finished and deployed a first set of fixes (see https://discord.com/channels/401396655599124480/978119396851589162/978619947561394206). Debugging and triaging issues reported by the team. Temporarily lifted requirement from the integration tests, as they are flaky. It should be finished by 2022-06-06.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue handling the developers' feedback, resolving issues. (Enso Bot - May 25, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-05-25):

**Progress:** Working towards acquisition of a new server to increase CI throughput, eventually making an order.
Looking into the issue where Ctrl+C left `cargo watch` processes running on Linux. Fixed the problem, while learning quite a bit along the way.
Looking into providing different option defaults depending on subcommand used. This is actually non-trivial, left for later set of fixes.
Fixed a regression where Project Manager packages failed to build on amd64 macOS. 
Improved diagnostics for case when downloaded CI run artifact archive cannot be extracted. It should be finished by 2022-06-06.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). below (Enso Bot - May 27, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-05-26):

**Progress:** Added GH API authentication to the build script calls on CI. This should fix the macOS build failures that were occurring from time to time. (Actually they were due to runner being GitHub-hosted, not really an OS-specific issue by itself.)
Renamed `clean` to `git-clean` as per suggestion to reduce risk of user accidently deleting unstaged work. 
Initial support for checking program versions (including `node`), as per build configuration file.
Support for checking compressed WASM size by specifying the limit in the build configuration file.
Whitelisting dependabot from CLA assistant and changelog checks, so PRs created by it are mergeable.
Fixing issue where wasm-pack-action (third party) randomly failed to recognize the latest version of wasm-pack (macOS runners), leading to failed builds.
Improve help for CI run source, to make clear that PAT token is required and what scope is expected there. It should be finished by 2022-06-06.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Deploy changes from today. Continue fixes. (Enso Bot - May 27, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-05-27):

**Progress:** Some initial post-acquire setup for the new CI server (setting up proper RAID).
Added support for caching artifacts downloaded through "<target>-source=ci-run".
Improving logs and diagnostics.
Added support for filtering logs through environment variable.
Started debugging issues where electron-build sometimes failed to build package. It should be finished by 2022-06-06.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). below (Enso Bot - May 31, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-05-30):

**Progress:** Preparing weekly presentation.
Spent tons of time debugging the electron-builder issue. It seems that it does not work properly with npm workspaces. Thinking it is missing dependencies, it deletes files it requires later on (by purging dev dependencies).
Done some work towards unifying release. It should be finished by 2022-06-06.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Finally deploy the updated build script. Either by deploying workaround to the electron-build issue or ripping out parts that trigger it.
Implement JUnit test report handling in the new CI jobs, as discussed with James. (Enso Bot - May 31, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-05-31):

**Progress:** Final set of fixes for the next build script deployment. Documenting workarounds for the electron-builder issue. General cleanups. Deploying the updated build script. It should be finished by 2022-06-06.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Work on unified nightly builds. (Enso Bot - Jun 2, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-06-01):

**Progress:** Started working on unified nightly builds. Implemented support for building Enso Engine through the new CLI. Done setup work on the new server that we bought for CI. Installed necessary software. Deployed provisional set of runners that will facilitate development of the new release pipeline. It should be finished by 2022-06-06.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Unified nightly builds. (Enso Bot - Jun 2, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-06-02):

**Progress:** Refactoring Engine building code to work nicely with GUI parts. Support for splitting backend build jobs (like building Proejct Manager and Engine independently). Refactoring CLI driver. It should be finished by 2022-06-06.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Unified nightly builds — continued. (Enso Bot - Jun 4, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-06-03):

**Progress:** Finished engine's cli driver cleanup (replacing argh with clap), deployed nightly builds workflow on the ci-build repository. It should be finished by 2022-06-06.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Fix issues, work on deploying to the staging repository. (Enso Bot - Jun 7, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for today (2022-06-06):

**Summary:** There is **14 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Huge scope. In general, progress is being tracked as part of weekly epic calls than this task. (Enso Bot - Jun 7, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-06-06):

**Progress:** Preparing and presenting weekly update.
Fixes for nightly builds on ci-builds repository, started working on workflow generation for enso-staging repository.
Doing some minor odd jobs and improvements in the build script. It should be finished by 2022-06-20.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Finish deployment to enso-staging. (Enso Bot - Jun 7, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-06-07):

**Progress:** Deployed the new nightly workflow on enso-staging repository. Development and fixes for its generation. 
Added pwsh to the Windows runner image. Regenerated all the Windows runners. It should be finished by 2022-06-20.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Build script support for Kaz's work on faster release builds. (Enso Bot - Jun 9, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-06-08):

**Progress:** Implementing support for manual handling of wasm-opt in the build script. Implementing logic for installing wasm-opt in the appropriate version.
Observed an issue with custom profile simulation through RUSTFLAGS. It should be finished by 2022-06-20.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Solve/Avoid the issue, update the script, merge PR. (Enso Bot - Jun 10, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-06-15):

**Progress:** Getting back to the nightlies. Updating and merging recent changes. Debugging and fixing issue with firebase configuration. Some fixes for the build script. It should be finished by 2022-06-20.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Hopefully the fix is sound. Then prepare a final iteration of the nightlies setup on the staging repository. (Enso Bot - Jun 17, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Friday (2022-06-17):

**Progress:** Significantly reworked the nightly release workflow, so different platforms do not have to wait for each other without need. Refactoring of the workflow generation code. It should be finished by 2022-06-20.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Rework the metarunner and deployment workflows to support multiple runner servers per platform. Include runners from the newly bought server in the general pool. (Enso Bot - Jun 19, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the last Saturday (2022-06-18):

**Progress:** Reworked metarunner logic, so we can have arbitrary number of servers for each platform. Each server has its own configuration (including an appropriate number of runners). Refreshed runner image.
Second Linux server runners joined the generally available pool, bringing up number of concurrent jobs from 4 to 9. It should be finished by 2022-06-20.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Deploy nightly ide builds to the main repository. Report electron-builder issue with npm workspaces. Bump the build script. (Enso Bot - Jun 21, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for yesterday (2022-06-20):

**Summary:** There is **2 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Additional work related to package layout and bugfixes for the workflow. (Enso Bot - Jun 22, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-06-21):

**Progress:** Deploying nightly releases to the main repository, which proved more troublesome than expected. Dmitry pointed out an issue with package layout, which I've addressed. Hubert reported unexpected failure, which might have been caused by race condition in CI workflow. Attempted a fix. Tons of cleanups and minor fixes related to the build script update that comes with nightly releases.

(This actually covers both Monday and Tuesday, these were like one extremely long day of fixes and waiting for builds.) It should be finished by 2022-06-22.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Finish deployment and follow with any necessary post-deploy fixes. Planning future work with Sylwia.
If nothing serious comes up, start esbuild task. (Enso Bot - Jun 22, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for today (2022-06-27):

**Summary:** There is **7 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Mergify bot incident. More issues appearing. Having time off last week. (Enso Bot - Jun 28, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for today (2022-06-27):

**Progress:** Weeklu presentation and update.
Debugging issues with new nightly-related workflows. Observed that GH git checkout v3 seems to also cause issues, downgraded to v2 for the time being. Debugging and fixing issue with Windows runner failing to properly handle non-unicode output from SBT. It should be finished by 2022-06-29.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue working towards deploying nightlies properly. (Enso Bot - Jun 28, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for the provided date (2022-06-22):

**Progress:** Preparing task descriptions for Sylwia.
Spent significant time on cleanup related to mergebot incident. Cancelling all the spurious runs programmatically, and waiting until CI dust settles.
Otherwise, I've continued to fix and debug various issues following the Engine CI update that goes together with nightly deployment. It should be finished by 2022-06-29. (Enso Bot - Jun 29, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-06-28):

**Progress:** Mostly just suffering of continuous poking stuff and seeing if it helps. Apparently eventually guessed the root problem. It should be finished by 2022-06-29.

**Next Day:** Next day I will be working on the [##180302898 task](https://www.pivotaltracker.com/story/show/180302898). Continue working towards deploying nightlies properly. (Enso Bot - Jun 30, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-06-29):

**Progress:** Working on interesting problem reported by Mateusz, where bad diagnostic was given in case where CI artifact upload failed due to GitHub being unavailable. 
Fixing issues with the Engine tests, with help of Dmitry. Final touches and updates on the nightlies/bump branch. It should be finished by 2022-06-29.

**Next Day:** Next day I will be working on the [##182604962 task](https://www.pivotaltracker.com/story/show/182604962). Resume esbuild task. (Enso Bot - Jun 30, 2022)
<hr />
**Michał Urbańczyk** reports a new **🔴 DELAY** for yesterday (2022-06-30):

**Summary:** There is **1 days** delay in implementation of the [GUI build script rewrite (#180302898) task](https://www.pivotaltracker.com/story/show/180302898).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Debugging and fixing final issue with engine tests took longer than expected. Than, when trying to test script, another issue appeared -- that, after debugging, turned out to be caused by temporary outage of npm registry. (Enso Bot - Jul 2, 2022)
<hr />
**Michał Urbańczyk** reports a new **STANDUP** for yesterday (2022-06-30):

**Progress:** Deployed the updated build script along and the new nightly release workflow. Documenting changes and doing final tests. Ordering new server/hardware upgrade in Hetzner. It should be finished by 2022-06-30.

**Next Day:** Next day I will be working on the [##182604962 task](https://www.pivotaltracker.com/story/show/182604962). Finish infrastructural work, resume esbuild task. (Enso Bot - Jul 2, 2022)
<hr />
